DPC-1071: Admin site separation: create new admin site

Sun-Mountain opened the pull request:
<!--

--- PR Hygiene Checklist ---

1. Make sure your branch is named with this format: `user-initials/description-ABC-123`. For example, `jj/add-awesomeness-bcda-99999`
2. Update the PR title: `dpc-99999 Feature: Add Awesomeness`
3. Edit the text below - do not leave placeholders in the text.
4. Add any other details that will be helpful for the reviewers: details description, screenshots, etc
5. Request a review from someone/multiple someones
-->

<!-- Replace xxx with the JIRA ticket number: -->

### Fixes [DPC-1071](https://jira.cms.gov/browse/DPC-1071)

<!-- Describe the problem being solved here: -->

This ticket kicks off the separation of the `internal_user` dashboard from the web app. We are creating a second Ruby on Rails app in the DPC repository in `dpc-admin`. This sets up the basic rails app structure and allows users to sign in via github.

### Proposed Changes

<!-- List of changes with bullet points here: -->

- Create `dpc-admin` in root `dpc-app`
- Create main layout and structure for admin portal
- Connect to existing web app database
- Users can log into a separate admin portal via github
- Tests for `internal_user` model and sign in.

### Change Details

<!-- Add detailed discussion of changes here: -->
- Had to change default `localhost` of `dpc-web` to 3500 so that one can run both admin dashboard and the web app simultaneously.
- Included instructions in both admin and web app readmes.

### Security Implications

<!-- Does the change deal with PII/PHI at all? What should reviewers look for in
terms of security concerns? -->

- [ ] new software dependencies

<!-- If yes, list the new dependencies and briefly note any relevant security impacts -->

- [ ] security controls or supporting software altered

<!-- If yes, what security controls or supporting software are affected? -->

- [ ] new data stored or transmitted

<!-- If yes, what new data are we storing or transmitting? Is the data considered PII/PHI? -->

- [ ] security checklist is completed for this change

<!-- If yes, provide a link to the security checklist in Confluence here. -->

- [ ] requires more information or team discussion to evaluate security implications

<!-- Use this to indicate you're unsure how this change may impact system security
and would like to solicit the team's feedback. Optionally, provide background
information regarding your questions and concerns. -->

- [x] no PHI/PII is affected by this change

<!-- If yes, provide what PHI/PII is affected by this change -->

### Acceptance Validation

<!-- Were you able to fully test the acceptance criteria on the related ticket? if not, why not? -->

- Users can log into a separate admin portal via github
- Create main layout and structure for admin portal

![Screen Shot 2021-02-17 at 11 25 10 AM](https://user-images.githubusercontent.com/44708048/108234527-e0374280-7112-11eb-997d-9d12a3d2c79c.png)

![Screen Shot 2021-02-17 at 11 24 49 AM](https://user-images.githubusercontent.com/44708048/108234539-e2999c80-7112-11eb-8e28-edaae84508b9.png)

- Have tests pass for internal user sign in and internal user model

![Screen Shot 2021-02-17 at 11 23 39 AM](https://user-images.githubusercontent.com/44708048/108234559-e9281400-7112-11eb-9630-7466ad0e05a0.png)

- Create README to set up local admin dashboard


<!-- Insert screenshots if applicable (drag images here) -->

### Feedback Requested

<!-- What type of feedback you want from your reviewers? -->

Any

PatKelsh reviewed (changes requested):
We should install rubocop and run "rubocop --auto-gen-config" or address the issues it raises. Also the ruby version for this is 2.6.3 but in dpc-web it is 2.7.1, should these be the same version to keep things consistent?
I don't see too many tests in the spec directory, there probably should be a bunch of test, especially for the base search service.

Sun-Mountain commented:
> We should install rubocop and run "rubocop --auto-gen-config" or address the issues it raises. Also the ruby version for this is 2.6.3 but in dpc-web it is 2.7.1, should these be the same version to keep things consistent?
> I don't see too many tests in the spec directory, there probably should be a bunch of test, especially for the base search service.

Good catch on the ruby version! And I've updated that and added rubocop.

The spec tests for base search will come when I work on the user/org dashboards. Right now the modals (that use the base search) aren't operational because I was struggling with the javascript.

PatKelsh reviewed (approved):
Looks good to me
